From e2623bb4c0099ce8fdffc901c05f55c5ab9a836f Mon Sep 17 00:00:00 2001
From: Daniel Gao <daniel.gao.work@gmail.com>
Date: Sun, 29 Jun 2025 15:08:54 -0400
Subject: [PATCH 1/4] Fix read before initialize of options.rtSettings.srgb

---
 rtgui/options.cc | 78 +++++++++++++++++++++++++-----------------------
 1 file changed, 41 insertions(+), 37 deletions(-)

diff --git a/rtgui/options.cc b/rtgui/options.cc
index 46d5f4edb..6683ce6c8 100644
--- a/rtgui/options.cc
+++ b/rtgui/options.cc
@@ -527,43 +527,6 @@ void Options::setDefaults()
     ICCPC_copyright = Options::getICCProfileCopyright();
     ICCPC_appendParamsToDesc = false;
 
-    fastexport_bypass_sharpening         = true;
-    fastexport_bypass_sharpenEdge        = true;
-    fastexport_bypass_sharpenMicro       = true;
-    //fastexport_bypass_lumaDenoise        = true;
-    //fastexport_bypass_colorDenoise       = true;
-    fastexport_bypass_defringe           = true;
-    fastexport_bypass_dirpyrDenoise      = true;
-    fastexport_bypass_dirpyrequalizer    = true;
-    fastexport_bypass_wavelet    = true;
-    fastexport_raw_bayer_method                  = "fast";
-    //fastexport_bypass_raw_bayer_all_enhance    = true;
-    fastexport_bypass_raw_bayer_dcb_iterations   = true;
-    fastexport_bypass_raw_bayer_dcb_enhance      = true;
-    fastexport_bypass_raw_bayer_lmmse_iterations = true;
-    fastexport_bypass_raw_bayer_linenoise        = true;
-    fastexport_bypass_raw_bayer_greenthresh      = true;
-    fastexport_raw_xtrans_method                 = "fast";
-    fastexport_bypass_raw_ccSteps        = true;
-    fastexport_bypass_raw_ca             = true;
-    fastexport_bypass_raw_df             = true;
-    fastexport_bypass_raw_ff             = true;
-    fastexport_icm_input_profile         = "(camera)";
-    fastexport_icm_working_profile       = "ProPhoto";
-    fastexport_icm_output_profile        = options.rtSettings.srgb;
-    fastexport_icm_outputIntent          = rtengine::RI_RELATIVE;
-    fastexport_icm_outputBPC             = true;
-    fastexport_resize_enabled            = true;
-    fastexport_resize_scale              = 1;
-    fastexport_resize_appliesTo          = "Cropped area";
-    fastexport_resize_method             = "Lanczos";
-    fastexport_resize_dataspec           = 3;
-    fastexport_resize_width              = 900;
-    fastexport_resize_height             = 900;
-    fastexport_resize_longedge           = 900;
-    fastexport_resize_shortedge          = 900;
-    fastexport_use_fast_pipeline         = true;
-
     clutsDir = "./cluts";
 
     cutOverlayBrush = std::vector<double> (4);
@@ -679,6 +642,45 @@ void Options::setDefaults()
 //  rtSettings.decaction =0.3;
 //  rtSettings.ciebadpixgauss=false;
     rtSettings.rgbcurveslumamode_gamut = true;
+
+    fastexport_bypass_sharpening         = true;
+    fastexport_bypass_sharpenEdge        = true;
+    fastexport_bypass_sharpenMicro       = true;
+    //fastexport_bypass_lumaDenoise        = true;
+    //fastexport_bypass_colorDenoise       = true;
+    fastexport_bypass_defringe           = true;
+    fastexport_bypass_dirpyrDenoise      = true;
+    fastexport_bypass_dirpyrequalizer    = true;
+    fastexport_bypass_wavelet    = true;
+    fastexport_raw_bayer_method                  = "fast";
+    //fastexport_bypass_raw_bayer_all_enhance    = true;
+    fastexport_bypass_raw_bayer_dcb_iterations   = true;
+    fastexport_bypass_raw_bayer_dcb_enhance      = true;
+    fastexport_bypass_raw_bayer_lmmse_iterations = true;
+    fastexport_bypass_raw_bayer_linenoise        = true;
+    fastexport_bypass_raw_bayer_greenthresh      = true;
+    fastexport_raw_xtrans_method                 = "fast";
+    fastexport_bypass_raw_ccSteps        = true;
+    fastexport_bypass_raw_ca             = true;
+    fastexport_bypass_raw_df             = true;
+    fastexport_bypass_raw_ff             = true;
+    fastexport_icm_input_profile         = "(camera)";
+    fastexport_icm_working_profile       = "ProPhoto";
+    // Must be set after options.rtSettings.srgb has already been initialized!
+    fastexport_icm_output_profile        = options.rtSettings.srgb;
+    fastexport_icm_outputIntent          = rtengine::RI_RELATIVE;
+    fastexport_icm_outputBPC             = true;
+    fastexport_resize_enabled            = true;
+    fastexport_resize_scale              = 1;
+    fastexport_resize_appliesTo          = "Cropped area";
+    fastexport_resize_method             = "Lanczos";
+    fastexport_resize_dataspec           = 3;
+    fastexport_resize_width              = 900;
+    fastexport_resize_height             = 900;
+    fastexport_resize_longedge           = 900;
+    fastexport_resize_shortedge          = 900;
+    fastexport_use_fast_pipeline         = true;
+
     lastIccDir = rtSettings.iccDirectory;
     lastDarkframeDir = rtSettings.darkFramesPath;
     lastFlatfieldDir = rtSettings.flatFieldsPath;
@@ -2873,6 +2875,8 @@ void Options::saveToFile(Glib::ustring fname)
 
 void Options::load(bool lightweight)
 {
+    // Start from fresh options
+    options = Options{};
 
     // Find the application data path
 
-- 
2.51.0

